### PR TITLE
use the mixpanel loaded callback and promises everywhere

### PR DESCRIPTION
**Pull request by emirotin**
Mixpanel is soo cool. For example, it doesn't fire the loaded callback if it is already loaded.
Spent literally half of the day hunting the issues like this.

This PR goes to the "main" PR branch.

Want you to overview the specific changes I made here.
Will then squash on the main PR branch after merging.

### CODE REVIEW

File: packages/resin-mixpanel-client/src/resin-mixpanel-client.js [hunk 1]
```diff
@@ -6,113 +7,124 @@ module.exports = function(token) {
...
+					if (!userId) {
+						throw new Error('(Resin Mixpanel Client) Please login() before using setUser()')
+					}
+					return mp.people.set(userId, props, callback)
+				})
+			})
 		},
-		track: function(event, properties, callback) {
-			var mp = mixpanel()
+		setUserOnce: function(props, callback) {
```
Comment (LucianBuzzo) [inline packages/resin-mixpanel-client/src/resin-mixpanel-client.js]: You can remove the `callback` parameter as it's unused and looks like it's always undefined.

File: packages/resin-mixpanel-client/src/resin-mixpanel-client.js [hunk 2]
```diff
@@ -6,113 +7,124 @@ module.exports = function(token) {
...
 
 	// the browser mixpanel library calls the callback with the response object (in verbose mode)
 	// or the status 0/1 in non-verbose mode
 	// we normalize it here to match the node style and work with Promise.fromCallback
-	function wrapCallback(callback) {
+	function wrapBrowserCallback(callback) {
 		if (!callback) return null
 		return function(response) {
-			if (response === 0) return callback(new Error('Mixpanel error'))
+			if (typeof response === 'number' && response !== 1) return callback(new Error('Mixpanel error: ' + response))
```
Comment (LucianBuzzo) [inline packages/resin-mixpanel-client/src/resin-mixpanel-client.js]: NIT: Can you user braces here?

File: packages/resin-universal-mixpanel/src/browser.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
-	init: function() {
-		window.mixpanel.init.apply(window.mixpanel, arguments)
-		// we have to wrap it into a getter because in the browser the global instance is replaced when the lib is loaded
-		return function () {
-			return window.mixpanel
-		}
+	init: function(token) {
+		return new Promise(function (resolve) {
+			var mixpanel = window.mixpanel
+			var isLoaded = mixpanel.__loaded
```
Comment (LucianBuzzo) [inline packages/resin-universal-mixpanel/src/browser.js]: You can remove `isLoaded` as it's never used.

File: packages/resin-universal-mixpanel/src/node.js [hunk 4]
```diff
@@ -1,12 +1,10 @@
...
 module.exports = {
-	init: function() {
-		var mixpanel = mixpanelLib.init.apply(mixpanelLib, arguments)
-		// we have to wrap it into a getter because in the browser the global instance is replaced when the lib is loaded
-		return function () {
-			return mixpanel
-		}
+	init: function(token) {
+		var mixpanel = mixpanelLib.init(token)
+		return  Promise.resolve(mixpanel)
```
Comment (LucianBuzzo) [inline packages/resin-universal-mixpanel/src/node.js]: NIT: Multiple spaces before `Promise`.

**Review by LucianBuzzo — APPROVED** [review]
Some NIT but otherwise LGTM

File: packages/resin-mixpanel-client/src/resin-mixpanel-client.js (hunk 1 above)
Comment (emirotin) [reply packages/resin-mixpanel-client/src/resin-mixpanel-client.js]: Right, it's a leftover

File: packages/resin-universal-mixpanel/src/browser.js (hunk 3 above)
Comment (emirotin) [reply packages/resin-universal-mixpanel/src/browser.js]: Ah, another one

**Review by pimterry — APPROVED** [review]

**Pull request merged**